framework/ChiMesh/SweepUtilities/AngleAggregation/angleaggregation.cc

```diff
@@ -49,8 +60,8 @@ void chi_mesh::sweep_management::AngleAggregation::ZeroIncomingDelayedPsi()
     {
       auto& rbndry = (BoundaryReflecting&)(*bndry);
 
-      if (rbndry.opposing_reflected)
-        for (auto& angle : rbndry.hetero_boundary_flux_old)
+      if (rbndry.IsOpposingReflected())
```

Comment:
I looked down two lines below what was shown in the github diff (that is, I looked at line 69) and I do not understand the val = 0. What are those nested loops actually doing?

Reply:
If the simulation has opposing reflected boundary conditions we essentially have a cycle there. One of the boundaries has to break the cycle, store delayed angular fluxes for 
- a particular angle,
- a particular cell,
- a particular face or faces on that cell (not all),
- a set of DOFs

and these need to be iterated on. Because the offsets here can be so irregular, we are just using a N-dimensional array. I am open to ideas if you can think of doing this in a simpler manner.